packages/grpc-js/src/server-call.ts

```diff
@@ -439,6 +442,56 @@ export class Http2ServerCallStream<
     return this.cancelled;
   }
 
+  private getDecompressedMessage(message: Buffer, encoding: string) {
+    switch (encoding) {
+      case 'deflate': {
+        return new Promise<Buffer | undefined>((resolve, reject) => {
+          zlib.inflate(message.slice(5), (err, output) => {
+            if (err) {
+              this.sendError({
+                code: Status.INTERNAL,
+                details: `Received "grpc-encoding" header "${encoding}" but ${encoding} decompression failed`,
+              });
+              resolve();
+            } else {
+              const joined = Buffer.concat([message.slice(0, 5), output]);
+              resolve(joined);
+            }
+          });
+        });
+      }
+  
+      case 'gzip': {
+        return new Promise<Buffer | undefined>((resolve, reject) => {
+          zlib.unzip(message.slice(5), (err, output) => {
+            if (err) {
+              this.sendError({
+                code: Status.INTERNAL,
+                details: `Received "grpc-encoding" header "${encoding}" but ${encoding} decompression failed`,
+              });
+              resolve();
+            } else {
+              const joined = Buffer.concat([message.slice(0, 5), output]);
+              resolve(joined);
+            }
+          });
+        });
+      }
+
+      case 'identity': {
+        return Promise.resolve(message);
+      }
+  
+      default: {
+        this.sendError({
+          code: Status.UNIMPLEMENTED,
+          details: `Received "grpc-encoding" header "${encoding}", which is not supported`,
```

Comment:
```suggestion
          details: `Received message compressed with unsupported encoding "${encoding}"`,
```

The server sends this error only when it receives a message that is actually compressed with an unsupported encoding, not just when it receives the header.

Reply:
True!

Reply:
True!